mlrun/utils/notifications/notification/slack.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+
+import requests
+
+import mlrun.utils.helpers
+
+from .base import NotificationBase
+
+
+class SlackNotification(NotificationBase):
+    emojis = {
+        "completed": ":smiley:",
+        "running": ":man-running:",
+        "error": ":x:",
+    }
+
+    def send(self):
+        webhook = self.params.get("webhook", None) or os.environ.get("SLACK_WEBHOOK")
+        if not webhook:
+            mlrun.utils.helpers.logger.warn(
+                "No slack webhook is set, skipping notification"
+            )
+            return
+
+        data = self._generate_slack_data()
+
+        response = requests.post(
+            webhook,
+            data=json.dumps(data),
+            headers={"Content-Type": "application/json"},
+        )
+        response.raise_for_status()
```

Comment:
if this is being called from the API, maybe we should add the actual client initialization in `mlrun.api.utils.clients`. wdyt?

Reply:
It can be called from both client and API so I think this isn't relevant